### PR TITLE
 added new functionaly on percentage click

### PR DESCRIPTION
**Pull request by lovelgeorge99**
https://qacc.atlassian.net/browse/QACC-421

### CODE REVIEW

File: src/components/DonatePage/DonatePageBody.tsx [hunk 1]
```diff
@@ -499,18 +501,38 @@ const DonatePageBody: React.FC<DonatePageBodyProps> = ({ setIsConfirming }) => {
...
         }
+
         return percentage;
       }
     });
   };
 
+  const handleRemainingCapClick = () => {
+    const remainingBalance = floor(tokenDetails?.formattedBalance);
+    if (remainingBalance >= userDonationCap) {
```
Comment (aminlatifi) [inline src/components/DonatePage/DonatePageBody.tsx:529]: I would use Math.min instead

File: src/components/DonatePage/DonatePageBody.tsx [hunk 2]
```diff
@@ -499,18 +501,38 @@ const DonatePageBody: React.FC<DonatePageBodyProps> = ({ setIsConfirming }) => {
...
-        const amount = floor((userDonationCap * percentage) / 100);
-        setInputAmount(amount.toString());
-        if (amount > parseFloat(tokenDetails.formattedBalance)) {
+        const remainingBalance = floor(tokenDetails?.formattedBalance);
+        const amount = floor((remainingBalance * percentage) / 100);
+        if (amount <= userDonationCap) {
+          setInputAmount(amount.toString());
+        } else if (amount > userDonationCap) {
+          setInputAmount(userDonationCap.toString());
+        }
```
Comment (aminlatifi) [inline src/components/DonatePage/DonatePageBody.tsx:510]:
```suggestion
        setInputAmount(Math.min(amount, userDonationCap).toString());
```

**Review by aminlatifi — CHANGES_REQUESTED** [review]
Thanks @lovelgeorge99 
Logic looks good, just few amendment

File: src/components/DonatePage/DonatePageBody.tsx (hunk 2 above)
Comment (lovelgeorge99) [reply src/components/DonatePage/DonatePageBody.tsx]: Thanks @aminlatifi i have updated the changes

**Review by ae2079 — APPROVED** [review]
thanks @lovelgeorge99 
LGTM

**Pull request merged**